app/controllers/user_controller.rb

```diff
@@ -0,0 +1,85 @@
+class UserController < ApplicationController
+    get '/login' do
+        if logged_in?
+            redirect to '/tweets'
+        end
+
+        session.delete(:login_error) ?
+            @login_error_message = session.delete(:login_error_message) :
+            @login_error_message = ''
+
+        erb :'/users/login'
+    end
+
+    post '/login' do
+        if params[:username].empty? 

```

Comment:
I think you could use [ActiveRecord validations](http://guides.rubyonrails.org/active_record_validations.html) (e.g. `validates_presence_of :username`) to perform these types of checks more easily.